ocean_provider/utils/util.py

```diff
@@ -218,9 +215,9 @@ def get_compute_info():
         return None, None
 
 
-def validate_order(web3, sender, token_address, num_tokens, tx_id, did, service_id):
+def validate_order(web3, sender, token_address, num_tokens, tx_id, did, service):
     logger.debug(
-        f"validate_order: did={did}, service_id={service_id}, tx_id={tx_id}, "
+        f"validate_order: did={did}, service_id={service.index}, tx_id={tx_id}, "
```

Comment:
service id should be kept for debugging, even if we are operating with the index on chain.

Reply:
I don't think there was a "service id" in v3.  In v3, `id` and `index` referred to the same thing.

Reply:
I originally reverted the changes to the `service_id` and added the `service_timeout` argument because we thought the previous changes were causing errors.  We now know the errors were from an external root cause.

I briefly considered "un-reverting" the changes to `service_id` again, but decided that I actually like the new `service_timeout` argument better.  It minimizes the data passed in - no need to pass the whole Service.